### PR TITLE
Write default data for pouch collection/variable storage keys

### LINKED ISSUE #2605: StorageStub.inflate error deserializing FindRestaurants Arc under PouchDb
Repro:

1. invoke web-shell with options `.../web-shell/?plannerStorage=volatile&storage=pouchdb://local/test`
2. type "restaurant" into search box
3. choose "Find restaurants" suggestion.
4. refresh the page

image

### PR DESCRIPTION
**Pull request by lindner**
Note: the pouch driver keeps a consistent view of the underlying data, so
starting an async write in the constructor is okay.

Fixes #2605

### CODE REVIEW

File: src/runtime/test/arc-tests.ts [hunk 1]
```diff
@@ -576,4 +576,57 @@ describe('Arc', () => {
     await rootSlotConsumer.contentAvailable;
     assert.equal(rootSlotConsumer._content.template, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ');
   });
+
+  ['volatile://', 'pouchdb://memory/user/'].forEach((storageKeyPrefix) => {
+    it('handles serialization/deserializatoin of empty arcs handles ' + storageKeyPrefix, async () => {
```
Comment (mykmartin) [inline src/runtime/test/arc-tests.ts]: s/deserializatoin/deserialization

File: src/runtime/storage/pouchdb/pouch-db-collection.ts [hunk 2]
```diff
@@ -42,6 +42,19 @@ export class PouchDbCollection extends PouchDbStorageProvider implements Collect
     super(type, storageEngine, name, id, key);
 
     this._model = new CrdtCollectionModel();
+
+    // Insure that the underlying database item is created.
```
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/storage/pouchdb/pouch-db-collection.ts]: thanks!

File: src/runtime/test/arc-tests.ts [hunk 3]
```diff
@@ -576,4 +576,57 @@ describe('Arc', () => {
...
+particle FavoriteFoodPicker in 'particles/Profile/source/FavoriteFoodPicker.js'
+  inout [FavoriteFood] foods
+  description \`select favorite foods\`
+    foods \`favorite foods\`
+
+recipe FavoriteFood
+  create #favoriteFoods as foods
+  FavoriteFoodPicker
+    foods = foods
+`, {loader, fileName: process.cwd() + '/input.manifest'});
```
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/test/arc-tests.ts]: done

File: src/runtime/storage/pouchdb/pouch-db-collection.ts [hunk 4]
```diff
@@ -42,6 +42,19 @@ export class PouchDbCollection extends PouchDbStorageProvider implements Collect
...
+    this.db.get(this.pouchDbKey.location).catch((err) => {
+      if (err.name === 'not_found') {
+        this.db.put({
+          _id: this.pouchDbKey.location,
+          model: this._model.toLiteral(),
+          referenceMode: this.referenceMode,
+          type: this.type.toLiteral()
+        }).catch((e) => {console.log('error init', e);});
```
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/storage/pouchdb/pouch-db-collection.ts]: done

File: src/runtime/test/arc-tests.ts (hunk 1 above)
Comment (lindner) [reply src/runtime/test/arc-tests.ts]: done

**Pull request merged**